incubator/exporter-otlp/src/main/java/io/aklivity/zilla/runtime/exporter/otlp/internal/serializer/OtlpMetricsDescriptor.java

```diff
@@ -0,0 +1,115 @@
+{
+    private static final Map<String, String> SERVER_METRIC_NAMES = Map.of(
+        "http.request.size", "http.server.request.size",
+        "http.response.size", "http.server.response.size",
+        "http.duration", "http.server.duration",
+        "http.active.requests", "http.server.active_requests"
+    );
+    private static final Map<String, String> CLIENT_METRIC_NAMES = Map.of(
+        "http.request.size", "http.client.request.size",
+        "http.response.size", "http.client.response.size",
+        "http.duration", "http.client.duration"
+    );
+    private static final Map<KindConfig, Map<String, String>> KIND_METRIC_NAMES = Map.of(
+        KindConfig.SERVER, SERVER_METRIC_NAMES,
+        KindConfig.CLIENT, CLIENT_METRIC_NAMES
+    );
+
+    private final Function<String, Metric> resolveMetric;
+    private final IntFunction<KindConfig> resolveKind;
+    private final Map<String, String> kinds;
+    private final Map<String, String> descriptions;
+    private final Map<String, String> units;
+
+    public OtlpMetricsDescriptor(
+        Function<String, Metric> resolveMetric,
+        IntFunction<KindConfig> resolveKind)
+    {
+        this.resolveMetric = resolveMetric;
+        this.resolveKind = resolveKind;
+        this.kinds = new Object2ObjectHashMap<>();
+        this.descriptions = new Object2ObjectHashMap<>();
+        this.units = new Object2ObjectHashMap<>();
+    }
+
+    public String kind(
+        String internalName)
+    {
+        String result = kinds.get(internalName);
+        if (result == null)
+        {
+            Metric.Kind kind = resolveMetric.apply(internalName).kind();
+            result = kind == COUNTER ? "sum" : kind.toString().toLowerCase();
+            kinds.put(internalName, result);
+        }
+        return result;
+    }
+
+    public String nameByBinding(
+        String internalMetricName,
+        int bindingId)
+    {
+        String result = null;
+        KindConfig kind = resolveKind.apply(bindingId);
+        Map<String, String> externalNames = KIND_METRIC_NAMES.get(kind);
+        if (externalNames != null)
+        {
+            result = externalNames.get(internalMetricName);
+        }
+        return result != null ? result : internalMetricName;
+    }
+
+    public String description(
+        String internalName)
+    {
+        String result = descriptions.get(internalName);
+        if (result == null)
+        {
+            result = resolveMetric.apply(internalName).description();
+            descriptions.put(internalName, result);
+        }
+        return result;
+    }
+
+    public String unit(
+        String internalName)
+    {
+        String result = units.get(internalName);
+        if (result == null)
+        {
+            Metric.Unit unit = resolveMetric.apply(internalName).unit();
+            result = unit == COUNT ? "" : unit.toString().toLowerCase();
+            units.put(internalName, result);
+        }
+        return result;
+    }
+}
```

Comment:
This `OtlpMetricsDescriptor` is tightly coupled to `OtlpMetricsSerializer`, so lets move it as an inner class of `OtlpMetricsSerializer`, and avoid needing to specify all the lambdas too.

Reply:
I refactored `OtlpMetricsDescriptor` to be an inner class of `OtlpMetricsSerializer`. I deleted the unit tests for now per your comments below. I will work on adding new IT's.